src/diffusers/models/modeling_utils.py

```diff
@@ -588,9 +684,46 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         else:
             if use_safetensors:
                 try:
+                    if not is_sharded:
+                        model_file = _get_model_file(
+                            pretrained_model_name_or_path,
+                            weights_name=_add_variant(SAFETENSORS_WEIGHTS_NAME, variant),
+                            cache_dir=cache_dir,
+                            force_download=force_download,
+                            resume_download=resume_download,
+                            proxies=proxies,
+                            local_files_only=local_files_only,
+                            token=token,
+                            revision=revision,
+                            subfolder=subfolder,
+                            user_agent=user_agent,
+                            commit_hash=commit_hash,
+                        )
+                    else:
+                        sharded_ckpt_cached_folder, sharded_metadata = _get_checkpoint_shard_files(
+                            pretrained_model_name_or_path,
+                            index_file,
+                            cache_dir=cache_dir,
+                            force_download=force_download,
+                            proxies=proxies,
+                            resume_download=resume_download,
+                            local_files_only=local_files_only,
+                            token=token,
+                            user_agent=user_agent,
+                            revision=revision,
+                            subfolder=subfolder,
+                            commit_hash=commit_hash,
+                        )
+
+                except IOError as e:
+                    if not allow_pickle:
+                        raise e
+                    pass
```

Comment:
I would add a log `logger.error("An error occurred while trying to fetch ...: {e}")` so that user knows something unusual has happened.
And then a `logger.warning("Defaulting to unsafe serialization. Pass `allow_pickle=False` to raise an error instead.")`.

(logs can be rephrased :smile:)